lib/ProductOpener/Images.pm

```diff
@@ -359,31 +358,12 @@ sub display_search_image_form($) {
 	my $product_image_with_barcode = $Lang{product_image_with_barcode}{$lang};
 	$product_image_with_barcode =~ s/( |\&nbsp;)?:$//;
 
-	$html .= <<HTML
-<div id="imgsearchdiv_$id">
-
-<a class="button small expand" id="imgsearchbutton_$id">@{[ display_icon('photo_camera') ]} $product_image_with_barcode
-<input type="file" accept="image/*" class="img_input" name="imgupload_search" id="imgupload_search_$id"
-	style="position: absolute;right:0;bottom:0;top:0;cursor:pointer;opacity:0;width:100%;height:100%;"/>
-</a>
-</div>
-
-<div id="progressbar_$id" class="progress" style="display:none">
-  <span id="progressmeter_$id" class="meter" style="width:0%"></span>
-</div>
-
-<div id="imgsearchmsg_$id" data-alert class="alert-box info" style="display:none">
-  $Lang{sending_image}{$lang}
-  <a href="#" class="close">&times;</a>
-</div>
-
-<div id="imgsearcherror_$id" data-alert class="alert-box alert" style="display:none">
-  $Lang{send_image_error}{$lang}
-  <a href="#" class="close">&times;</a>
-</div>
-
-HTML
-;
+	my $template_data_ref = {
+		product_image_with_barcode => $product_image_with_barcode,
+		id => $id,
+		lang => \&lang,
```

Comment:
lang and display_icon are now added automatically by process_template(), so we don't need them here

Reply:
done!